lib/server.go

```diff
@@ -2045,10 +2047,25 @@ func (srv *Server) _processTransactions(pp *Peer, transactions []*MsgDeSoTxn) []
 	// a block. Doing something like this would make it so that if a transaction
 	// was initially rejected due to us not having its dependencies, then we
 	// will eventually add it as opposed to just forgetting about it.
-	glog.V(2).Infof("Server._processTransactions: Processing %d transactions from "+
+	glog.V(1).Infof("Server._processTransactions: Processing %d transactions from "+
 		"peer %v", len(transactions), pp)
 	transactionsToRelay := []*MempoolTx{}
-	for _, txn := range transactions {
+	for ii, txn := range transactions {
+		// Take some time to allow other threads to get the ChainLock if they need it
+		//
+		// TODO: It's not obvious how necessary this rest period is, and it's also not obvious if
+		// five seconds is the right amount. We added it during a mission-critical sprint
+		// to find and fix slow block production issue as one of several patches. It clearly doesn't
+		// hurt so we decided to leave it in for now.
+		if (ii+1)%1000 == 0 {
```

Comment:
Early on, I suspected that the reason get-block-template wasn't working was because of a ChainLock starvation caused by downloading a lot of transactions. Ultimately, it wasn't that, but giving a breather for other processes to acquire the ChainLock seemed like a fine thing to keep, and doesn't slow down sync too much.